src_features/networkDynamic/network_dynamic.c

```diff
@@ -0,0 +1,644 @@
+    }
+
+    // Handle the payload
+    sw = handle_next_icon_chunk(data, length);
+    if (sw != APDU_RESPONSE_OK) {
+        return sw;
+    }
+    if (g_icon_payload.received_size == g_icon_payload.expected_size) {
+        // Everything has been received
+        sw = parse_icon_buffer();
+    }
+    return sw;
+}
+#endif
+
+/**
+ * @brief Parse the SIGNATURE value.
+ *
+ * @param[in] data buffer received
+ * @param[in] field_len Length of the field value
+ * @param[in] sig_ctx the signature context
+ * @return APDU Response code
+ */
+static uint16_t parse_signature(const uint8_t *data, uint16_t field_len, s_sig_ctx *sig_ctx) {
+    sig_ctx->sig_size = field_len;
+    sig_ctx->sig = data;
+    return APDU_RESPONSE_OK;
+}
+
+/**
+ * @brief Verify the payload signature
+ *
+ * Verify the SHA-256 hash of the payload against the public key
+ *
+ * @param[in] sig_ctx the signature context
+ * @return whether it was successful
+ */
+static bool verify_signature(s_sig_ctx *sig_ctx) {
+    uint8_t hash[INT256_LENGTH];
+    cx_err_t error = CX_INTERNAL_ERROR;
+    bool ret_code = false;
+
+    CX_CHECK(
+        cx_hash_no_throw((cx_hash_t *) &sig_ctx->hash_ctx, CX_LAST, NULL, 0, hash, INT256_LENGTH));
+
+    CX_CHECK(check_signature_with_pubkey("Dynamic Network",
+                                         hash,
+                                         sizeof(hash),
+                                         LEDGER_SIGNATURE_PUBLIC_KEY,
+                                         sizeof(LEDGER_SIGNATURE_PUBLIC_KEY),
+#ifdef HAVE_LEDGER_PKI
+                                         CERTIFICATE_PUBLIC_KEY_USAGE_COIN_META,
+#endif
+                                         (uint8_t *) (sig_ctx->sig),
+                                         sig_ctx->sig_size));
+
+    ret_code = true;
+end:
+    return ret_code;
+}
+
+/**
+ * @brief Print the registered network.
+ *
+ * Only for debug purpose.
+ */
+static void print_network_info(void) {
+    char chain_str[sizeof(uint64_t) * 2 + 1] = {0};
+
+    PRINTF("****************************************************************************\n");
+    u64_to_string(DYNAMIC_NETWORK_INFO[g_current_slot].chain_id, chain_str, sizeof(chain_str));
+    PRINTF("[NETWORK] - Registered in slot %d: %s (%s), for chain_id %s\n",
+           g_current_slot,
+           DYNAMIC_NETWORK_INFO[g_current_slot].name,
+           DYNAMIC_NETWORK_INFO[g_current_slot].ticker,
+           chain_str);
+}
+
+/**
+ * @brief Parse the TLV payload containing the Network configuration.
+ *
+ * @param[in] data buffer received
+ * @param[in] length of the buffer
+ * @return APDU Response code
+ */
+static uint16_t parse_tlv(const uint8_t *data, uint8_t length) {
+    uint16_t offset = 0;
+    uint16_t field_tag = 0;
+    uint16_t field_len = 0;
+    uint16_t sw = APDU_RESPONSE_INTERNAL_ERROR;
+    uint16_t tag_start_off;
+    s_sig_ctx sig_ctx = {0};
+
+    // Set the current slot
+    g_current_slot = (g_current_slot + 1) % MAX_DYNAMIC_NETWORKS;
```

Comment:
This should be put at the end of the function, in the same `if` as the call to `print_network_info` so that we can ensure the first index to be used is 0, then 1, and so on. :+1:

Reply:
Need to keep it here, at the begining of the function, to ensure the icon, sent separately will use the same slot.
We could use another slot_index, but the code would be less readable.